src/hugr/rewrite.rs

```diff
@@ -0,0 +1,60 @@
+//! Rewrite operations on the HUGR - replacement, outlining, etc.
+
+#[allow(clippy::module_inception)] // TODO: Rename?
+pub mod replace;
+pub mod simple_replace;
+use std::mem;
+
+use crate::Hugr;
+pub use replace::{OpenHugr, Replace, ReplaceError};
+pub use simple_replace::{SimpleReplacement, SimpleReplacementError};
+
+/// An operation that can be applied to mutate a Hugr
+pub trait Rewrite<E> {
```

Comment:
Each rewrite has an specific error type. `E` should be an associated type (`Error` ?) instead of a generic parameter.

Reply:
Yes, good call, thanks